diff --git a/voltha/extensions/omci/state_machines/performance_intervals.py b/voltha/extensions/omci/state_machines/performance_intervals.py
index e9c70c5..d36dcbb 100644
--- a/voltha/extensions/omci/state_machines/performance_intervals.py
+++ b/voltha/extensions/omci/state_machines/performance_intervals.py
@@ -340,6 +340,8 @@ class PerformanceIntervals(object):
         config = self._device.configuration
 
         for pon in config.ani_g_entities:
+            if pon is None:
+                continue
             entity_id = pon['entity-id']
             self.delete_pm_me(FecPerformanceMonitoringHistoryData.class_id, entity_id)
             self.delete_pm_me(FecPerformanceMonitoringHistoryData.class_id, entity_id)
diff --git a/voltha/extensions/omci/tasks/alarm_check_task.py b/voltha/extensions/omci/tasks/alarm_check_task.py
index d62d6ee..371ecaa 100644
--- a/voltha/extensions/omci/tasks/alarm_check_task.py
+++ b/voltha/extensions/omci/tasks/alarm_check_task.py
@@ -31,7 +31,7 @@ class AlarmDataTask(Task):
     name = "Alarm Data Task"
     max_payload = 29
 
-    def __init__(self, omci_agent, device_id, class_id, entity_id):
+    def __init__(self, omci_agent, device_id):
         """
         Class initialization
 
@@ -46,8 +46,6 @@ class AlarmDataTask(Task):
                                             priority=AlarmDataTask.task_priority,
                                             exclusive=False)
         self._local_deferred = None
-        self._class_id = class_id
-        self._entity_id = entity_id
         self._last_number_of_commands = None
 
     def cancel_deferred(self):
@@ -81,8 +79,7 @@ class AlarmDataTask(Task):
         """
         Sync the current alarm sequence number
         """
-        self.log.info('perform-get-interval', class_id=self._class_id,
-                      entity_id=self._entity_id)
+        self.log.info('perform-get-alarm-data')
 
         try:
             device = self.omci_agent.get_device(self.device_id)
